Clarified and included CJK requirements in large text definition and updated rules affected by this change (#2121)
act-rules/act-rules.github.io: pages/glossary/large-scale-text.md

Reviewer: I rewrote a few things, including updating the links as I found this a little difficult to read with all the URLs in them:

```md
**Exception**: If the text nodes contain Unicode characters in the 4E00–9FFF range:

- at least 22 [points][], or
- at least 18 [points][] and a [computed][] [font weight][] CSS property of 700 or higher.

**Notes:**  
The WCAG definition of [large scale text](https://www.w3.org/TR/WCAG21/#dfn-large-scale) includes an exception for CJK (Chinese, Japanese, and Korean) text. The Unicode range 4E00–9FFF includes these characters. The "equivalent size" numbers are not included in WCAG, and come from the [Official Japanese Translation of WCAG 2.1](https://waic.jp/translations/WCAG21/#dfn-large-scale).  This translation offers the following insight (translated from Japanese to English to allow everyone understanding the specs):  

> TRANSLATION NOTE  
> In the case of Japanese full-width characters, 22 point or 18 point bold type is considered to be the "equivalent" size, based on the First Report of the Enlarged Textbook Promotion Council, "Chapter 2 Standards for Enlarged Textbooks." is reasonable."
Assuming all CJK characters pose similar contrast challanges, this definition extends the Japanese requirement to the whole CJK characters set included in the [Unicode standard](https://unicode.org/charts/PDF/U4E00.pdf) within the range of 4E00–9FFF.

[computed]: https://www.w3.org/TR/css-cascade-3/#computed-value
[font-size]: https://www.w3.org/TR/css-fonts-3/#propdef-font-size
[points]: https://www.w3.org/TR/css-values/#pt
[font weight]: https://www.w3.org/TR/css-fonts-3/#font-weight-prop
```

Reviewer 2: Taiwan has different requirements than Japan, I propose the following:

```
**Exception**:  
A text node containing Unicode characters in the 4E00–9FFF range ([CJK Unified Ideographs](https://unicode.org/charts/PDF/U4E00.pdf)) is <dfn id="large-scale-text:cjk">large scale text</dfn> if at least one of the following is true:

- the Japanese text node [computed][] [font-size][] is at least 22 [points][], or
- the Traditional Chinese text node [computed][] [font-size][] is at least 20 [points][], or
- the Japanese text node [computed][] [font-size][] is at least 18 [points][] and has a [computed][] [font weight][] of 700 or higher, or
- the Traditional Chinese text node [computed][] [font-size][] is at least 16 [points][] and has a [computed][] [font weight][] of 700 or higher.

**Notes:**  
The WCAG definition of [large scale text](https://www.w3.org/TR/WCAG21/#dfn-large-scale) includes an exception for CJK (Chinese, Japanese, and Korean) fonts. The Unicode range 4E00–9FFF includes these characters. The "equivalent size" numbers are not included in WCAG, and come from the [Taiwanese Readability Reference Guidelines](https://crpd.sfaa.gov.tw/BulletinCtrl?func=downloadFile&type=file&id=3749&code=F9FEF1F8F1FAF6FAFCFAFCFCFFFAF2F3) (PDF download) and the [Unofficial Japanese Translation of WCAG 2.1](https://waic.jp/translations/WCAG21/#dfn-large-scale). The Taiwanese Readability Reference Guidelines requires minimal font size 16 points for Traditional Chinese characters and 14 points for English characters to be fulfill the need of the people with disabilities, and the Unofficial Japanese translation offers the following insight (translated from Japanese to English to allow everyone understanding the specs):
```

  - DOM tree
---

A text node is large scale text if at least one of the following is true:

- the text node [computed][] [font-size][] is at least 18 [points][], or
- the text node [computed][] [font-size][] is at least 14 [points][] and has a [computed][] [font weight][] of 700 or higher.

> **Notes:**  

Reviewer 3: ```suggestion
**Background:**  
```

In the domain of languages such as Chinese, Japanese, and Korean (CJK languages), encompassing the Unicode character range from 4E00 to 9FFF ([CJK Unified Ideographs](https://unicode.org/charts/PDF/U4E00.pdf)), it becomes crucial to acknowledge the intricate nuances in their typographic requirements. Despite sharing a common Unicode spectrum, each CJK language and its corresponding country may exhibit distinct typographic preferences and standards.

Particularly noteworthy is the lack of uniformity in defining text sizes as either large or small within the following ranges:
- from 18 [points][] to 22 [points][]
- from 14 [points][] to 18 [points][] with a [computed][] [font weight][] of 700 or higher.

> Given the diversity in these specifications, the ACT rules adhere to the large text definition outlined above. This definition is applicable across all languages under the rationale of "no false positives."

Reviewer 3: ```suggestion
Whether CJK characters in these range should be considered as "large scale" depends on the language of the text, as well as the country for which the text is designed and whose requirements must be followed. Given that these pieces of information are not readily available, the ACT rules adhere to the large text definition outlined above. This definition is applicable across all languages under the rationale of "no false positives."
```

  - DOM tree
---

Text nodes with a [computed](https://www.w3.org/TR/css-cascade-3/#computed-value) [font-size][] of:

- at least 18 [points](https://www.w3.org/TR/css-values/#pt), or
- at least 14 [points](https://www.w3.org/TR/css-values/#pt) and a [computed](https://www.w3.org/TR/css-cascade-3/#computed-value) [font weight][] CSS property of 700 or higher.

**Exception**:  
If the text nodes contain Unicode characters in the 4E00–9FFF range ([CJK Unified Ideographs](https://unicode.org/charts/PDF/U4E00.pdf)), these characters are:

- at least 22 [points][], or

> - at least 18 [points][] and a [computed][] [font weight][] CSS property of 700 or higher.

Reviewer 3: **Suggestion:** I think we should rephrase the full definition (as it doesn't look like correct English now), in the line of:

> A text node is _large scale text_ if its [computed][] [font-size][] is:
> - at least 18 [points[]; or
> - at least 14 [points][] and its [computed][] [font+weight][] is at least 700.
>
> Exception:
> A text node containing CJK characters is _large scale text_ if …

Maybe even start the definition with "depending whether the text node contains CJK characters, …" 🤔 

Also, if the defined term instances (`_large scale text_`) are wrapped in a `dfn` element, we can reference them directly from the running text:
> `<dfn id="large-scale-text:cjk">large scale text</dfn>`
> `This is [large scale text][#large-scale-text:cjk]`

See, e.g. [attribute value in image button name rule](https://github.com/act-rules/act-rules.github.io/blob/develop/_rules/image-button-non-empty-accessible-name-59796f.md?plain=1#L182)

Additionally (but not directly related to this PR, so may be taken separately):
* These lists do not match how we write condition lists now; should be "if at least one of the following is true: the [computed][] [font-size][] is at least 18 points; or…"
* Technically, text nodes do not have CSS properties, only their parent element has, so the language used here is a bit incorrect (although not really confusing…)
* Also, not need to say "[font weight][] **CSS propert**" when the name already links to CSS specs.

Author: Suggestions applied rewriting the definition and starting with "A text node..."

---

> Additionally (but not directly related to this PR, so may be taken separately):
> 
> These lists do not match how we write condition lists now; should be "if at least one of the following is true: the [computed][] [font-size][] is at least 18 points; or…"

Happy to edit condition lists and include them in the current edits.

> Technically, text nodes do not have CSS properties, only their parent element has, so the language used here is a bit incorrect (although not really confusing…)

You're absolutely right. This was the case in the previous iteration as well. I'm struggling to find a straightforward method to enhance its clarity without adding more complexity to the definition.

> Also, not need to say "[font weight][] CSS propert" when the name already links to CSS specs.

updated.

**Notes:**  

> The WCAG definition of [large scale text](https://www.w3.org/TR/WCAG21/#dfn-large-scale) includes an exception for CJK (Chinese, Japanese, and Korean) fonts. The Unicode range 4E00–9FFF includes these characters. The "equivalent size" numbers are not included in WCAG, and come from the [Official Japanese Translation of WCAG 2.1](https://waic.jp/translations/WCAG21/#dfn-large-scale).  This translation offers the following insight (translated from Japanese to English to allow everyone understanding the specs):  

Reviewer 3: Isn't that an *un*official translation?

Author: Missed it while accepting Wilco suggestion. Good catch.
Updated

> TRANSLATION NOTE  

> > In the case of Japanese full-width characters, 22 point or 18 point bold type is considered to be the "equivalent" size, based on the First Report of the Enlarged Textbook Promotion Council, "Chapter 2 Standards for Enlarged Textbooks." is reasonable.

Reviewer 4: that final

`is reasonable.`

looks out of place. a remnant from a previous version of that sentence?

Reviewer 4: ```suggestion
> In the case of Japanese full-width characters, 22 point or 18 point bold type is considered to be the "equivalent" size, based on the First Report of the Enlarged Textbook Promotion Council, "Chapter 2 Standards for Enlarged Textbooks."
```

- at least 18 [points](https://www.w3.org/TR/css-values/#pt), or
- at least 14 [points](https://www.w3.org/TR/css-values/#pt) and a [computed](https://www.w3.org/TR/css-cascade-3/#computed-value) [font weight](https://www.w3.org/TR/css-fonts-3/#font-weight-prop) CSS property of 700 or higher.

> For CJK (Chinese, Japanese and Korean) characters with a [computed](https://www.w3.org/TR/css-cascade-3/#computed-value) [font-size](https://www.w3.org/TR/css-fonts-3/#propdef-font-size) of: 

Reviewer 5: ```suggestion
For CJK (Chinese, Japanese and Korean) characters within text nodes, a [computed](https://www.w3.org/TR/css-cascade-3/#computed-value) [font-size](https://www.w3.org/TR/css-fonts-3/#propdef-font-size) of: 
```

**Notes:**  
The WCAG definition of [large scale text](https://www.w3.org/TR/WCAG21/#dfn-large-scale) lacks clarity regarding the equivalent [font-size](https://www.w3.org/TR/css-fonts-3/#propdef-font-size) for CJK characters. According to WCAG:  
>[large scale (text)](https://www.w3.org/TR/WCAG21/#dfn-large-scale): with at least 18 point or 14 point bold or font size that would yield equivalent size for Chinese, Japanese and Korean (CJK) fonts

Since there is no official WCAG definition of **equivalent size** for CJK characters, we rely on the guidelines provided in the [Japanese translation of the WCAG 2.1](https://waic.jp/translations/WCAG21/#dfn-large-scale). This document offers the following insight (translated from Japanese to English to allow everyone understanding the specs):  

>TRANSLATION NOTE  
>In the case of Japanese full-width characters, 22 point or 18 point bold type is considered to be the "equivalent" size, based on the First Report of the Enlarged Textbook Promotion Council, "Chapter 2 Standards for Enlarged Textbooks." is reasonable."

> Assuming all CJK characters pose similar contrast challanges, this definition extends the Japanese requirement to the whole CJK characters set included in the [Unicode standard](https://unicode.org/charts/PDF/U4E00.pdf) within the range of 4E00–9FFF.

Reviewer 5: We might want to indicate that that Unicode range describes the CJK Unified Ideographs

Author: Added at the beginning of the Exception for a better understanding

- the text node [computed][] [font-size][] is at least 14 [points][] and has a [computed][] [font weight][] of 700 or higher.

> **Background:**  

Reviewer 3: ```suggestion
#### Background
```
Let's make it into a section.
